src/graph_notebook/neptune/client.py

```diff
@@ -0,0 +1,498 @@
+        if headers is None:
+            headers = {}
+
+        s = SPARQLWrapper('')
+        s.setQuery(query)
+        query_type = s.queryType.upper()
+        if query_type in ['SELECT', 'CONSTRUCT', 'ASK', 'DESCRIBE']:
+            return self.sparql_query(query, headers, explain, path=path)
+        else:
+            return self.sparql_update(query, headers, explain, path=path)
+
+    # TODO: enum/constants for supported types
+    def sparql_explain(self, query: str, explain: str = 'dynamic', output_format: str = 'text/html',
+                       headers=None, path: str = '') -> requests.Response:
+        if headers is None:
+            headers = {}
+
+        if 'Accept' not in headers:
+            headers['Accept'] = output_format
+
+        return self.sparql(query, headers, explain, path=path)
+
+    def sparql_status(self, query_id: str = ''):
+        return self._query_status('sparql', query_id=query_id)
+
+    def sparql_cancel(self, query_id: str, silent: bool = False):
+        if type(query_id) is not str or query_id == '':
+            raise ValueError('query_id must be a non-empty string')
+        return self._query_status('sparql', query_id=query_id, silent=silent, cancelQuery=True)
+
+    def get_gremlin_connection(self) -> client.Client:
+        uri = f'{self._http_protocol}://{self.host}:{self.port}/gremlin'
+        request = self._prepare_request('GET', uri)
+
+        ws_url = f'{self._ws_protocol}://{self.host}:{self.port}/gremlin'
+        ws_request = httpclient.HTTPRequest(ws_url, headers=dict(request.headers))
+        return client.Client(ws_request, 'g')
+
+    def gremlin_query(self, query, bindings=None):
+        c = self.get_gremlin_connection()
+        try:
+            result = c.submit(query, bindings)
+            future_results = result.all()
+            results = future_results.result()
+            c.close()
+            return results
+        except Exception as e:
+            c.close()
+            raise e
+
+    def gremlin_http_query(self, query, headers=None) -> requests.Response:
+        if headers is None:
+            headers = {}
+
+        uri = f'{self._http_protocol}://{self.host}:{self.port}/gremlin'
+        data = {'gremlin': query}
+        req = self._prepare_request('POST', uri, data=json.dumps(data), headers=headers)
+        res = self._http_session.send(req)
+        return res
+
+    def gremlin_status(self, query_id: str = '', include_waiting: bool = False):
+        kwargs = {}
+        if include_waiting:
+            kwargs['includeWaiting'] = True
+        return self._query_status('gremlin', query_id=query_id, **kwargs)
+
+    def gremlin_cancel(self, query_id: str):
+        if type(query_id) is not str or query_id == '':
+            raise ValueError('query_id must be a non-empty string')
+        return self._query_status('gremlin', query_id=query_id, cancelQuery=True)
+
+    def gremlin_explain(self, query: str) -> requests.Response:
```

Comment:
More a question than a comment. Not sure what current best practices are but the profile and explain methods are identical except for the endpoint. Is there any value to merging them so that future fixes (if any) do not have to be applied in 2 places?

Reply:
We could probably combine them (and SPARQL explain) into an `_plan` method that can configure and call for the result appropriately